### PR TITLE
Include sequencing to analysis files linking ebi-ait/dcp-ingest-central#210

### PR DESCRIPTION
**Pull request by aaclan-ebi**
ebi-ait/dcp-ingest-central#210

### CODE REVIEW

File: tests/exporter/graph/test_graph_crawler.py [hunk 1]
```diff
@@ -42,9 +43,12 @@ def test_generate_complete_experiment_graph(self):
...
-        test_project = MetadataResource.from_dict(self.files.get_entity('projects', 'mock-project'))
+        test_assay_process = MetadataResource.from_dict(self.mock_files.get_entity('processes', 'mock-assay-process'))
+        test_project = MetadataResource.from_dict(self.mock_files.get_entity('projects', 'mock-project'))
         experiment_graph = crawler.generate_experiment_graph(test_assay_process, test_project)
 
-        self.assertEqual(len(experiment_graph.nodes.get_nodes()), 15)
-        self.assertEqual(len(experiment_graph.links.get_links()), 4)  # 3 process links and 1 supplementary files link
+        expected_links = self.mock_files.get_links_json()
+
+        self.assertEqual(len(experiment_graph.nodes.get_nodes()), 18)
```
Comment (amnonkhen) [inline tests/exporter/graph/test_graph_crawler.py]: Where does this 18 come from?

File: tests/exporter/graph/test_graph_crawler.py [hunk 2]
```diff
@@ -42,9 +43,12 @@ def test_generate_complete_experiment_graph(self):
...
+        self.assertEqual(len(experiment_graph.links.get_links()), 5)  # 3 process links and 1 supplementary files link
```
Comment (amnonkhen) [inline tests/exporter/graph/test_graph_crawler.py]: Seems like a stale comment: 5 in the code, 3 and 1 in the comment.

**Review by amnonkhen — CHANGES_REQUESTED** [review]

File: exporter/graph/graph_crawler.py [hunk 3]
```diff
@@ -65,7 +71,20 @@ def _crawl(self, process: MetadataResource) -> ExperimentGraph:
         processes_to_crawl = GraphCrawler.flatten([self.metadata_service.get_derived_by_processes(i) for i in process_info.inputs])
 
         return reduce(lambda g1, g2: g1.extend(g2),
-                      map(lambda proc: self._crawl(proc), processes_to_crawl),
+                      map(lambda proc: self._crawl_up(proc), processes_to_crawl),
+                      partial_graph)
+
+    def _crawl_down(self, process: MetadataResource) -> ExperimentGraph:
```
Comment (amnonkhen) [inline exporter/graph/graph_crawler.py]: `_crawl_up` and `_crawl_down` look very similar apart from the input to `GraphCrawler.flatten`. Have you considered extracting this into a common function?

File: exporter/graph/graph_crawler.py [hunk 4]
```diff
@@ -33,7 +33,13 @@ def generate_experiment_graph(self, process: MetadataResource, project: Metadata
         return experiment_process_graph.extend(supplementary_files_graph)
 
     def experiment_graph_for_process(self, process: MetadataResource) -> ExperimentGraph:
-        return self._crawl(process)
+        upper_graph = self._crawl_up(process)
```
Comment (amnonkhen) [inline exporter/graph/graph_crawler.py]: Why do you use "upper" and "lower" instead of "inputs" and"derived"?

**Review by amnonkhen — CHANGES_REQUESTED** [review]

**Review by amnonkhen — DISMISSED** [review]

**Review by yusra-haider — DISMISSED** [review]

File: exporter/graph/graph_crawler.py (hunk 4 above)
Comment (aaclan-ebi) [reply exporter/graph/graph_crawler.py]: That's a good suggestion!

File: tests/exporter/graph/test_graph_crawler.py (hunk 1 above)
Comment (aaclan-ebi) [reply tests/exporter/graph/test_graph_crawler.py]: I've updated the assertions to check the actual uuid's from links.json. Please check.

**Review by jacobwindsor — APPROVED** [review]

**Review by yusra-haider — APPROVED** [review]

**Review by amnonkhen — APPROVED** [review]

**Pull request merged**